FreeTubeApp/FreeTube
# Fix trending & search result list rendered items not updated without workaround

## Pull Request Type


## Related issue



This **includes** changes from #3341

## Description

Similar to #3341 but with playlist and channel data entries handled
If there is a fourth data type this might be broken

## Screenshots 

Lazy!

## Testing 



Follow #3341 **first**

Test with both Local/Invidious API plz
- Search whatever keyword
- Switch to different types to warm up search result cache
- Switch to different types to ensure rendered search result list updated 

## Desktop

- **OS:**
- **OS Version:**
- **FreeTube version:**

## Additional context

This can replace #3341 if merged before it
